classes/local/datasource/site.php

```diff
@@ -61,4 +57,47 @@ public function get_data(): \stdClass {
     public static function is_optional(): bool {
         return false;
     }
+
+    /**
+     * Constructs the data structure to act as the data source. Uses a cache.
+     *
+     * @return \stdClass
+     * @throws \coding_exception
+     */
+    public function get_cached_data(): \stdClass {
+        // Because the data is constant, and small, using a cache would be counter-productive.
+        return $this->get_data();
+    }
+
+    /**
+     * Returns the cache key fragment for the instance data.
+     * If null, then caching should be avoided, both here and for the overall instance.
+     *
+     * @return string|null
+     */
+    public function get_instance_cache_key(): ?string {
+        // Returns a constant, because caching is not used in this datasource.
+        return '';
```

Comment:
If we're not caching, shouldn't this return null?

Reply:
No, because it's not vetoing. This is a special case.